test/fluttertagger_test.dart

```diff
@@ -0,0 +1,272 @@
+    });
+
+    testWidgets('triggers search callback with correct query', (tester) async {
+      String searchQuery = '';
+      String triggerChar = '';
+      onSearch = (query, triggerCharacter) {
+        searchQuery = query;
+        triggerChar = triggerCharacter;
+      };
+
+      testWidget = MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: FlutterTagger(
+              overlay: Container(
+                height: 100,
+                color: Colors.grey,
+                child: const Center(child: Text('Overlay')),
+              ),
+              controller: controller,
+              onSearch: onSearch,
+              builder: (context, key) {
+                return TextField(
+                  key: key,
+                  controller: controller,
+                  onChanged: (value) {},
+                );
+              },
+              triggerCharacterAndStyles: const {
+                '@': TextStyle(color: Colors.blue),
+                '#': TextStyle(color: Colors.green),
+              },
+            ),
+          ),
+        ),
+      );
+
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(searchQuery, 'test');
+      expect(triggerChar, '@');
+    });
+
+    testWidgets('displays overlay when typing a trigger character',
+        (tester) async {
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(find.text('Overlay'), findsOneWidget);
+    });
+
+    testWidgets('formats and displays tagged text correctly', (tester) async {
+      await tester.pumpWidget(testWidget);
+      await tester.enterText(find.byType(TextField), '@testUser');
+      await tester.pump();
+
+      expect(controller.text, '@testUser');
+    });
+
+    testWidgets('hides overlay when exiting search context', (tester) async {
```

Comment:
It would be better if this test can verify that the overlay is first shown before verifying that entering a character that terminates the search context causes the overlay to be dismissed.

Reply:
Updated the test

Reply:
This doesn't look updated